samples: drivers: watchdog running on the stm32H7 platforms

FRASTM opened the pull request:
Change the condition to have the samples/drivers/watchdog PASSED on the stm32h7 target boards.

FRASTM committed:
samples: drivers: watchdog running on the stm32H7 platforms

Change the condition to have the samples/drivers/watchdog PASSED
on the stm32h7 target boards.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

martinjaeger commented on samples/drivers/watchdog/boards/stm32h7_wwdg.overlay:
```diff
@@ -11,10 +11,6 @@
 	};
 };
 
-&rcc {
-	d2ppre1 = <16>;
-};
-
```
Can you please state in the commit message why the RCC setting was wrong? You added it a while ago in commit 838cd28195631e039b569f39cdc10cf1c839807f saying it is required.

FRASTM replied:
You're right
 even if at that time, the sysclock was much lower (typ. 96MHz) for the stm32h7xx, (now it is 480) there is no need to change it.

FRASTM committed:
drivers: watchdog: stm32 wwdg enable Early WKUP int with pclk

Enable the early WakeUp Interrupt at init, once the WWDG instance
is correclty cloked. Else deadlock can occur.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on samples/drivers/watchdog/sample.yaml:
```diff
@@ -48,7 +48,19 @@ tests:
     extra_args: DTC_OVERLAY_FILE=boards/stm32h7_wwdg.overlay
     filter: dt_compat_enabled("st,stm32-window-watchdog")
-    platform_allow: nucleo_h743zi
+    platform_allow:
+      - nucleo_h723zg
+      - nucleo_h745zi_q
+      - stm32h7s78_dk
+      - nucleo_h743zi
+      - stm32h735g_disco
+      - nucleo_h753zi
+      - stm32h750b_dk
+      - stm32h7b3i_dk
+      - stm32h745i_disco
+      - nucleo_h755zi_q
+      - stm32h747i_disco
+      - nucleo_h7a3zi_q
```
What about filtering on `SOC_SERIES_STM32H7X` ?

FRASTM replied:
`filter: dt_compat_enabled("st,stm32-window-watchdog") and CONFIG_SOC_SERIES_STM32H7X`
does not filter correctly
neither
`filter: CONFIG_SOC_SERIES_STM32H7X and CONFIG_WWDG_STM32`

FRASTM commented:
keep rcc D2PPR1 prescaler  unchanged for the stm32H7 serie.
In the stm32 WWDg driver, enable the watchdog IRQ at the end of the initialization process, once the peripheral is clocked
else it might cause lock (seen on nucleo_h743zi) :
```
*** Booting Zephyr OS build (tainted) v3.7.0-5097-gdfbcea777a10 ***                                                               
Watchdog sample application                                                                                                       
Attempting to test pre-reset callback                                                                                             
[00:00:00.011,000] <dbg> wdt_wwdg_stm32: wwdg_stm32_install_timeout: prescaler: 64                                                
[00:00:00.019,000] <dbg> wdt_wwdg_stm32: wwdg_stm32_install_timeout: Desired WDT: 100000 us                                       
[00:00:00.028,000] <dbg> wdt_wwdg_stm32: wwdg_stm32_install_timeout: Set WDT:     100488 us                                       
Feeding watchdog 5 times                                                                                                          
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Waiting for reset...      
```

nashif commented on samples/drivers/watchdog/sample.yaml:
```diff
@@ -48,7 +48,18 @@ tests:
     extra_args: DTC_OVERLAY_FILE=boards/stm32h7_wwdg.overlay
     filter: dt_compat_enabled("st,stm32-window-watchdog")
-    platform_allow: nucleo_h743zi
+    platform_allow:
```
isnt this already covered with the filter above? why the platform allow thing?

nashif replied:
also, please add integration_platforms with one platform for PR verification.

erwango replied:
One typical example where https://github.com/zephyrproject-rtos/zephyr/issues/81458 would help.
